akka-stream-tests/src/test/scala/akka/stream/scaladsl/FlowSectionSpec.scala

```diff
@@ -19,23 +23,37 @@ class FlowSectionSpec extends AkkaSpec(FlowSectionSpec.config) {
 
   "A flow" can {
 
-    "have an op with a name" in {
-      val n = "Converter to Int"
-      val f = Flow[Int].section(name(n))(_.map(_.toInt))
-      f.toString should include(n)
+    "have an op with a different dispatcher" in {
+      val flow = Flow[Int].section(dispatcher("my-dispatcher1"))(_.map(sendThreadNameTo(testActor)))
+
+      Source.single(1).via(flow).to(Sink.ignore).run()
+
+      expectMsgType[String] should include("my-dispatcher1")
     }
 
-    "have an op with a different dispatcher" in {
-      val flow = Flow[Int].section(dispatcher("my-dispatcher"))(_.map(sendThreadNameTo(testActor)))
+    "have a nested flow with a different dispatcher" in {
+      val flow = Flow[Int].section(dispatcher("my-dispatcher1"))(_.via(Flow[Int].map(sendThreadNameTo(testActor))))
 
       Source.single(1).via(flow).to(Sink.ignore).run()
 
-      receiveN(1).foreach {
-        case s: String ⇒ s should include("my-dispatcher")
-      }
+      expectMsgType[String] should include("my-dispatcher1")
+    }
+
+    "have multiple levels of nesting" in {
+      val flow = Flow[Int].section(dispatcher("my-dispatcher1"))(
```

Comment:
we can remove section actually. Calling withAttributes on a Flow, then using simple via can work as an alternative.